pulp_rpm/app/kickstart/treeinfo.py

```diff
@@ -275,42 +276,40 @@ def images(self):
 
         Returns:
             list: List of image data
-
         """
         if self._images:
             return self._images
 
-        images = []
-        temp = {}
+        # self._image_paths is {path: [platform-list]
+        images = []  # list-of {name: str, path: str, platforms: [platform-list]}
+        temp = defaultdict(set)  # {stanza: set(paths)}
 
         for key in self._data.keys():
-            if key.startswith("images"):
+            if key.startswith("images"):  # e.g. images-xen or images-x86_64
                 image_key = key
-                platform = image_key.split("-")[1]
-                for key, value in self._data.get(image_key, {}).items():
-                    temp.update({key: value})
-
-                    _platform = platform
-                    if value in self._image_paths.keys():
-                        _platform = f"{self._image_paths[value]}, {platform}"
-
-                    self._image_paths.update({value: _platform})
-
-        for key, value in temp.items():
-            image = {}
-            image["name"] = key
-            image["path"] = value
-            image["platforms"] = self._image_paths[value]
-
-            self._image_paths[value] = {}
-
-            images.append(image)
+                platform = image_key.split("-")[1]  # e.g. xen or x86_64
+                for stanza, path in self._data.get(image_key, {}).items():
+                    # A given stanza can have multiple paths
+                    if path:
```

Comment:
`path` can be `None` then?  or empty string

Reply:
The root problem we're trying to solve here, is that "temp" was a dict of {stanza: path}. In RHEL5, you can have multiple paths for initrd and kernel images (xen and pxe) - so we're now storing path as a set, and creating an image name/path/platform instance for each/all of them.

See the following lines in the diff for the important changes:

- https://github.com/pulp/pulp_rpm/pull/2491/files#diff-9ea371735ec5b37aff3c3d927a4b579f44a166748728b9b4dd434e3ff38de399R285
- https://github.com/pulp/pulp_rpm/pull/2491/files#diff-9ea371735ec5b37aff3c3d927a4b579f44a166748728b9b4dd434e3ff38de399R294
- https://github.com/pulp/pulp_rpm/pull/2491/files#diff-9ea371735ec5b37aff3c3d927a4b579f44a166748728b9b4dd434e3ff38de399R302-R303

Reply:
> `path` can be `None` then? or empty string

This is inherited from the original code, and I think is just to let the .items() call work even when image_key doesn't exist in self._data. I think I can rewrite the loop-processing to clear up what's going on.